AMW_angular/io/src/app/apps/app-add/app-add.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, EventEmitter, Input, Output, Signal } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+import { NgSelectModule } from '@ng-select/ng-select';
+import { Release } from '../../settings/releases/release';
+import { Release as Rel } from '../../resource/release';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { Resource } from '../../resource/resource';
+import { AppCreate } from '../app-create';
+
+@Component({
+  selector: 'app-app-add',
+  standalone: true,
+  imports: [FormsModule, NgSelectModule],
+  templateUrl: './app-add.component.html',
+})
+export class AppAddComponent {
+  @Input() releases: Signal<Release[]>;
+  @Input() appServerGroups: Signal<Resource[]>;
+  @Output() saveApp: EventEmitter<AppCreate> = new EventEmitter<AppCreate>();
+
+  app: AppCreate = { appName: '', appReleaseId: null, appServerId: null, appServerReleaseId: null };
+  appServerGroup: Resource;
+  appServerRelease: Rel;
+
+  constructor(public activeModal: NgbActiveModal) {
+    this.activeModal = activeModal;
+  }
+
+  hasInvalidGroup(): boolean {
+    const isInvalid =
+      this.appServerGroup === undefined || this.appServerGroup === null || this.appServerGroup?.releases.length === 0;
+    if (isInvalid) {
+      this.appServerRelease = undefined;
```

Comment:
this seems to be a side effect of this method. Why is this necessary?

Reply:
This resets the ng-select for appServerReleases in case the appServer has been reset/unselected